src/vault/client/api_util.clj

```diff
@@ -6,13 +6,29 @@
     [clojure.tools.logging :as log]
     [clojure.walk :as walk])
   (:import
+    (clojure.lang
+      ExceptionInfo)
     (java.security
       MessageDigest)
     (org.apache.commons.codec.binary
       Hex)))
 
 ;; ## API Utilities
 
+(defmacro support-not-found
+  "Tries to perform the api call, and if a 404 occurs, looks for :not-found in on-fail-opts"
```

Comment:
Probably want to add "looks for and returns the value of `:not-found` in `on-fail-opts` if present." to be more explicit about its behavior.